homeassistant/components/energy/data.py

```diff
@@ -0,0 +1,108 @@
+"""Energy data."""
+from __future__ import annotations
+
+from typing import Optional, TypedDict, cast
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import singleton, storage
+
+from .const import DOMAIN
+
+STORAGE_VERSION = 1
+STORAGE_KEY = DOMAIN
+
+
+@singleton.singleton(DOMAIN)
+async def async_get_manager(hass: HomeAssistant) -> EnergyManager:
+    """Return an initialized data manager."""
+    manager = EnergyManager(hass)
+    await manager.async_initialize()
+    return manager
+
+
+class EnergyData(TypedDict):
+    """Dictionary holding the energy data."""
+
+    # This is a continues increasing value
+    stat_house_energy_meter: str | None
+
+    stat_solar_generatation: str | None
+    stat_solar_return_to_grid: str | None
+    stat_solar_predicted_generation: str | None
```

Comment:
Typo noted by @jesserockz, it's `generation`

Also, @Danielhiversen noted that `solar` is an implementation detail and it can be wind or batteries too.

Reply:
Can we support a list of energy producers (solar array, wind, etc.?)